torch/csrc/jit/serialization/export.h

```diff
@@ -53,6 +55,42 @@ TORCH_API std::string serialize_model_proto_to_string(
 
 TORCH_API void check_onnx_proto(const std::string& proto_string);
 
+// Serializer for both oldsyle and unified format TS serialization
+class ScriptModuleSerializer {
+ public:
+  explicit ScriptModuleSerializer(
+      caffe2::serialize::PyTorchStreamWriter& export_writer)
+      : writer_(export_writer) {}
+
+  void writeFiles(const std::string& code_dir);
+  void serialize(
+      const Module& module,
+      const ExtraFilesMap& extra_files,
+      bool bytecode_format,
+      bool save_mobile_debug_info);
+
+  ~ScriptModuleSerializer() = default;
+
+ private:
+  void convertNamedType(const c10::NamedTypePtr& class_type);
+  void convertTypes(const at::NamedTypePtr& root_type);
+  void writeExtraFiles(const Module& module, const ExtraFilesMap& extra_files);
+  void writeMobileMetadata(
+      const Module& module,
+      const ExtraFilesMap& extra_files);
+  void writeByteCode(const Module& module, bool save_mobile_debug_info);
+  void writeArchive(const IValue& value, const std::string& archive_name);
+
+  caffe2::serialize::PyTorchStreamWriter& writer_;
```

Comment:
Changed from `caffe2::serialize::PyTorchStreamWriter writer_` since the next PR in the stack can only pass in a reference